### DISCUSSION TITLE
STUdio pack format

### DISCUSSION THREAD

**Discussion by Oripy** [marian-m12l/studio#314] [Q&A]
Hello,

Is there a place that describes how a STUdio pack is structured? I have made a small tool to quickly chop down a file into "intro" and "story" parts to use on some files containing ads and unnecessary "blabla", and I would also like to automate the pack creation.

Where can I find the description of the structure of a STUdio pack?

**Comment by vpoulailleau** [comment]
You may like https://github.com/jersou/studio-pack-generator to generate the packs.

**Reply by Oripy** [reply]
Thank you for the link but I don't really want to install a whole DENO environment just to put some files into a folder (if the pack generation is easy enough). As far as I understand, the pack looks like it is just a bunch of files (audio and images) with names encoded as SHA1 with a JSON file that put everything together. 
I could retro-engineer the format, but with open-source software, I kind of hoped not to have to go that way.
I would just need a basic understanding of the STUdio pack format, I do not need a full documentation.

**Reply by vpoulailleau** [reply]
I started to write such tool (with retro-engineering), and in the JSON there are many internal references to other parts of the JSON. I didn't find documentation, but maybe it exists. I was almost there, but then you have to encode/improve the sound, adjust images, generate titles and so on (https://github.com/jersou/studio-pack-generator#features). I didn't have time to re-invent the wheel and that DENO tool does exactly what I wanted to code, so I used it. But I agree that it's a "big" tool…
